eclass/cargo.eclass: specify --path . to install

cardoe opened the pull request:
cargo install has long required --path . for 2018 edition crates but not
required it for 2015 edition crates. It is supported however for 2015
edition crates and works for all versions in the tree so it makes sense
to make it the default.

Closes: https://bugs.gentoo.org/703590
Signed-off-by: Doug Goldstein <cardoe@gentoo.org>

gyakovlev commented:
@gentoo/rust 

duplicating what've left of the bug:

passing --path=. breaks some crates with meta-crates, look at alacritty as an example.

I have to pass --path=alacritty

if we pass both --path=. and --path=alacritty cargo blows up.

meta crates are not that common, and probably I can do 'cd alacritty' before running cargo or re-define functions, but this needs a bit of additional investigation.

gyakovlev commented:
```
>>> Completed testing x11-terms/alacritty-0.4.0

>>> Install x11-terms/alacritty-0.4.0 into /var/tmp/portage/x11-terms/alacritty-0.4.0/image
error: The argument '--path <PATH>' was provided more than once, but cannot be used multiple times

```

cardoe commented:
Fixed in the latest push. Thanks for testing that out. Added `CARGO_INSTALL_PATH` as an eclass variable that defaults to `.` but for alacritty we can override it.

gyakovlev commented:
```
/home/ya/src/gentoo/eclass/cargo.eclass: line 40: .: filename argument required
.: usage: . filename [arguments]
```
some unwanted bash magic happens here.

gyakovlev commented:
probably it should be
`: ${CARGO_INSTALL_PATH:=.}`

gyakovlev commented:
also I think it makes sense to define `CARGO_INSTALL_PATH` in `src_install` in `alacritty` ebuilds, but I can edit this myself. that makes it a bit more obvious that's it's a special snowflake, global scope vars can be easily overlooked and forgotten.

also not sure if it's a great name
cargo already uses and sets some envvars
https://doc.rust-lang.org/cargo/reference/environment-variables.html
I'd change it to `ECARGO_INSTALL_PATH` just for sake of future-proofing it against name clashes.

gyakovlev commented:
ok all the affected ebuilds work fine with my fix, `: ${CARGO_INSTALL_PATH:=.}`
yes, I really built them all =D

merging to unblock myself on future work, I'll open a PR. 
it will include unconditional --offline for everything, configurable verbosity (some users do not like `-vv` and it will honor `NOCOLOR` setting in make.conf, it was often requested to unclobber logs from color sequences.

I did not rename `CARGO_INSTALL_PATH` to ECARGO, do as you wish.

gyakovlev commented:
https://doc.rust-lang.org/cargo/reference/config.html
`Environment variables

Cargo can also be configured through environment variables in addition to the TOML syntax above. For each configuration key above of the form foo.bar the environment variable CARGO_FOO_BAR can also be used to define the value. For example the build.jobs key can also be defined by CARGO_BUILD_JOBS.
`
another way of possible clashes, so eclass variables should avoid starting with `CARGO_`
